src/Apps/Collect/Routes/Collection/Components/Header/DefaultHeader.tsx

```diff
@@ -1,181 +1,130 @@
-                    context_page_owner_id: collection_id,
-                    context_page_owner_slug: collection_slug,
-                    destination_path: artwork.href,
-                  })
-                }}
-              >
-                <HeaderImage
-                  height={defaultHeaderImageHeight}
-                  src={
-                    smallViewport
-                      ? (artwork.image.small.url as string)
-                      : (artwork.image.large.url as string)
-                  }
-                  preventRightClick
-                />
-              </a>
-            )
-          })}
-        </HeaderArtworks>
-      </DefaultHeaderContainer>
-    </header>
+    <Box my={[0, 0, 2]}>
+      <Media greaterThanOrEqual="md">
+        <Rail>
+          {largeArtworks.map(artwork => (
+            <DefaultHeaderArtwork
+              key={artwork.node.id}
+              artwork={artwork}
+              collectionId={collectionId}
+              collectionSlug={collectionSlug}
+            />
+          ))}
+        </Rail>
+      </Media>
+
+      <Media lessThan="md">
+        <Rail>
+          {smallArtworks.map(artwork => (
+            <DefaultHeaderArtwork
+              key={artwork.node.id}
+              artwork={artwork}
+              collectionId={collectionId}
+              collectionSlug={collectionSlug}
+              small
+            />
+          ))}
+        </Rail>
+      </Media>
+    </Box>
   )
 }
 
-export const getHeaderArtworks = (
-  artworksArray: any[],
-  headerWidth: number,
-  isSmallViewport: boolean
-) => {
-  let artworkWidths = 0
-  let shouldAppendDuplicateArtworksToHeader = true
-  const headerArtworks: any[] = []
-
-  if (artworksArray.length < 1) {
-    return [] as any[]
-  }
-
-  /**
-   * Loop through the initial artworks array, appending an artwork to the output array,
-   * until the summed widths of the artworks in the output array are greater than the
-   * width of the viewport.
-   */
-  while (shouldAppendDuplicateArtworksToHeader) {
-    for (const { node: artwork } of artworksArray) {
-      if (artworkWidths > headerWidth) {
-        headerArtworks.push(artwork)
-        shouldAppendDuplicateArtworksToHeader = false
-        return headerArtworks
-      }
-
-      headerArtworks.push(artwork)
-
-      isSmallViewport
-        ? (artworkWidths += artwork.image.small.width + IMAGE_MARGIN_X)
-        : (artworkWidths += artwork.image.large.width + IMAGE_MARGIN_X)
-    }
-  }
-}
-
 export const CollectionDefaultHeaderFragmentContainer = createFragmentContainer(
   CollectionDefaultHeader,
   {
     headerArtworks: graphql`
       fragment DefaultHeader_headerArtworks on FilterArtworksConnection {
         edges {
           node {
-            href
-            slug
+            id
             image {
-              small: resized(height: 160) {
-                url
+              # TODO: Should accept arguments
```

Comment:
I need to look over the docs on fragment args. In practice this isn't an issue since if the parent passes different args Relay errors, but...